opm/input/eclipse/Schedule/UDQ/UDQActive.cpp

```diff
@@ -16,211 +16,308 @@
-}
+        if (UDQ::well_control(record.control)) {
+            loadRstWellUDAs(record, wgIndex, uda, well_names, udaRecords);
+        }
+        else if (UDQ::is_group_production_control(record.control)) {
+            loadRstGroupProdUDAs(record, wgIndex, uda, group_names, udaRecords);
+        }
+        else {
+            loadRstGroupInjUDAs(record, wgIndex, rst_active.ig_phase,
+                                uda, group_names, udaRecords);
+        }
+    }
 
-std::string UDQActive::wg_hash(const std::string& wgname, UDAControl control) {
-    return wgname + std::to_string(static_cast<int64_t>(control));
+    return udaRecords;
 }
 
-/*
-  We go through the current list of input records and compare with the supplied
-  arguments (uda, wgnamem, control). There are seven possible outcomes:
-
-    0. The uda variable is not defined (defaulted target/limit): fast return.
-
-    1. The uda variable is a double and no uda usage has been registered so far:
-       fast return.
-
-    2. The uda variable is a double, and the (wgname,control) combination is
-       found in the input data; this implies that uda has been used previously
-       for this particular (wgname, control) combination: We remove that record
-       from the input_data.
-
-    3. The uda variable is a string and we find that particular (udq, wgname,
-       control) combination in the input data: No changes
-
-    4. The uda variable is a string; but another udq was used for this (wgname,
-       control) combination: We erase the previous entry and add a new entry.
-
-    5. The uda variable is a string and we do not find this (wgname, control)
-       combination in the previous records: We add a new record.
-
-    6. The uda variable is a double, and the (wgname, control) combination has
-       not been encountered before: return 0
+Opm::UDQActive::operator bool() const
+{
+    return ! this->input_data.empty();
+}
 
-*/
-int UDQActive::update(const UDQConfig& udq_config, const UDAValue& uda, const std::string& wgname, UDAControl control) {
+// We go through the current list of input records and compare with the supplied
+// arguments (uda, wgnamem, control).  There are seven possible outcomes:
+//
+//   0. The uda variable is not defined (defaulted target/limit): fast return.
+//
+//   1. The uda variable is a double and no uda usage has been registered so far:
+//      fast return.
+//
+//   2. The uda variable is a double, and the (wgname,control) combination
+//      is found in the input data; this implies that uda has been used
+//      previously for this particular (wgname, control) combination: We
+//      remove that record from the input_data.
+//
+//   3. The uda variable is a string and we find that particular (udq,
+//      wgname, control) combination in the input data: No changes
+//
+//   4. The uda variable is a string; but another udq was used for this
+//      (wgname, control) combination: We erase the previous entry and add a
```

Comment:
replace

Reply:
Thanks.  I've pushed an update to that effect.